src/http/reply.cc

```diff
@@ -133,20 +133,28 @@ void reply::write_body(const sstring& content_type, sstring content) {
 }
 
 future<> reply::write_reply(output_stream<char>& out) {
-    add_header("Transfer-Encoding", "chunked");
-    return out.write(response_line()).then([this, &out] () mutable {
-        return write_reply_headers(out);
-    }).then([&out] () mutable {
-        return out.write("\r\n", 2);
-    }).then([this, &out] () mutable {
-        if (_skip_body) {
-            return make_ready_future<>();
+    return out.write(response_line().data(), response_line().size()).then([this, &out] {
```

Comment:
In the `! if (_resp->body_writer)` branch the `resp->_response_line` was written, not `response_line()`. I assume they are the same at this point, but now the `_response_line` becomes obsoleted and can be removed? (can be done incrementally)